File: ios/Classes/SDK/SdkRegionMapping.swift
import Gateway

// `GatewayRegion` alone refers to the Pigeon enum of this plugin; the SDK type is qualified
// with its module name.
func toSdkRegion(_ region: GatewayRegion) -> Gateway.GatewayRegion {
  switch region {
  case .mtf: return .mtf
  case .europe: return .europe
  case .northAmerica: return .northAmerica
  case .asiaPacific: return .asiaPacific
  case .india: return .india
  case .china: return .china
  case .saudiArabia: return .saudiArabia
  }
}

File: android/src/main/kotlin/com/example/nbe_payment_flutter_plugin/sdk/SdkRegionMapping.kt
package com.example.nbe_payment_flutter_plugin.sdk

import com.example.nbe_payment_flutter_plugin.generated.GatewayRegion
import com.mastercard.gateway.android.sdk.GatewayRegion as SdkGatewayRegion

internal fun toSdkRegion(region: GatewayRegion): SdkGatewayRegion = when (region) {
    GatewayRegion.MTF -> SdkGatewayRegion.MTF
    GatewayRegion.EUROPE -> SdkGatewayRegion.EUROPE
    GatewayRegion.NORTH_AMERICA -> SdkGatewayRegion.NORTH_AMERICA
    GatewayRegion.ASIA_PACIFIC -> SdkGatewayRegion.ASIA_PACIFIC
    GatewayRegion.INDIA -> SdkGatewayRegion.INDIA
    GatewayRegion.CHINA -> SdkGatewayRegion.CHINA
    GatewayRegion.SAUDI_ARABIA -> SdkGatewayRegion.SAUDI_ARABIA
}
